hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestRegionServerReportForDuty.java

```diff
@@ -241,6 +243,136 @@ public void run() {
     waitForClusterOnline(master);
   }
 
+  /**
+   * Tests that the RegionServer's reportForDuty gets rejected by the master when the master is
+   * configured to reject decommissioned hosts and when there is a match for the joining
+   * RegionServer in the list of decommissioned servers. Test case for HBASE-28342.
+   */
+  @Test
+  public void testReportForDutyGetsRejectedByMasterWhenConfiguredToRejectDecommissionedHosts()
+    throws Exception {
+    // Start a master and wait for it to become the active/primary master.
+    // Use a random unique port
+    cluster.getConfiguration().setInt(HConstants.MASTER_PORT, HBaseTestingUtil.randomFreePort());
```

Comment:
Is this necessary? Shouldn't it be the default behavior for the test harness? If you have to do it for the master, why don't you have to do it for the region servers?

Reply:
I tried to stick to the same approach of starting master servers in this test suite, you can see it in `testReportForDutyWithMasterChange` and `testReportForDutyWithEnvironmentEdge`.